Gemfile

```diff
@@ -42,6 +42,7 @@ gem 'bugsnag-capistrano', '< 2', require: false
 # https://github.com/nahi/httpclient/issues/369
 gem 'httpclient', github: 'mikz/httpclient', branch: 'ssl-env-cert'
 gem 'oauth2'
+gem 'k8s-client'
```

Comment:
Any reason you preferred not to use https://github.com/abonas/kubeclient?

Reply:
I think @unleashed suggested using this one. I don't really have a preference.

Reply:
As you wish. https://github.com/3scale/porta is using https://github.com/abonas/kubeclient. @abonas is Alissa Bonas, Red Hatter, BTW.

Reply:
Hm. Looks I got it confused and picked a wrong one :)

Reply:
@guicassolato yes I am :)
The library is currently fantastically maintained by @cben , also a Red Hatter

Reply:
Ooh, a 3rd client :tada:, I wasn't aware of this one.
(the other one I knew is https://github.com/kubernetes-client/ruby,  documented "WIP" though picking up development recently)

Reply:
@cben they're all here :) https://kubernetes.io/docs/reference/using-api/client-libraries/#community-maintained-client-libraries

Reply:
Thanks, @abonas and @cben. We've been successfully using the gem in 3scale for OpenShift to discover manageable services deployed onto the same cluster.

Reply:
I'm going to stick with k8s-client for now. We are on a pretty tight schedule. It's `K8s::Client.autoconfig` is a very easy way for us to get a working client. I tried to do the same with kubeclient, but Config & Context do not expose namespace. I don't really have the time now to contribute, but will revisit it in the future.

Reply:
The one I suggested is k8s-client, yes, after attending their talk @ KubeCon EU 2019, which I recommend anyone interested (both users & maintainers) to watch. I remembered there was another one by a redhatter (now I see it's @abonas' & @cben's one) but got the impression it was not actively maintained, which it would seem is not quite true. :)